src/proxy/inbound.rs

```diff
@@ -244,10 +245,12 @@ impl Inbound {
                     SocketAddr::new(loopback, ri.upstream_addr.port()),
                 )
             } else {
-                (
-                    enable_original_source.then_some(ri.rbac_ctx.conn.src.ip()),
-                    ri.upstream_addr,
-                )
+                let upstream_src_ip = if pi.disable_inbound_freebind {
```

Comment:
If we don't need this logic we should delete it. Also a comment somewhere should explain _why_ we want to opt-out of original src for the new usecase

Reply:
Our CNI plugin configures captured pods so that ztunnel may freebind in their netns. It doesn't configure capture for the ztunnel pods though. For ztunnel itself, freebinds are blocked blocked by the kernel. The connection just times out if we try to freebind_connect to ourself.